src/Controller/Profile.php

```diff
@@ -0,0 +1,76 @@
+<?php
+namespace App\Controller;
+
+use App\Entity\User as UserEntity;
+use App\Service\User as UserService;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
+use Symfony\Component\HttpFoundation\Request;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+
+
+class Profile extends AbstractController
+{
+    /**
+     * @var UserService $userService
+     */
+    private $userService;
+
+    /**
+     * Construct.
+     */
+    public function __construct(
+        UserService $userService
+    )
+    {
+        $this->userService = $userService;
+    }
+
+    /**
+     *
+     */
+    public function index(Request $request, UserPasswordEncoderInterface $encoder)
+    {
+        $form = $this->createFormBuilder()
+            ->add('password', RepeatedType::class, array(
+                'type' => PasswordType::class,
+                'invalid_message' => 'Senhas precisam ser idênticas.',
+                'options' => array('attr' => ['class' => 'password-field']),
+                'required' => true,
+                'first_options'  => ['label' => 'Senha'],
+                'second_options' => ['label' => 'Repetir senha'],
+            ))
+            ->add('submit', SubmitType::class, ['label' => 'Atualizar perfil'])
+            ->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->handleEditFormSubmission($form, $encoder);
+        }
+
+        return $this->render('profile/index.html.twig', array(
+            'form' => $form->createView(),
+        ));
+    }
+
+    /**
+     * 
+     */
+    public function handleEditFormSubmission($form, $encoder)
+    {
+        $data = $form->getData();
+        
+        $this->userService->updatePassword($this->getUser(), $data['password'], $encoder);
+
+        $this->addFlash(
+            'success',
```

Comment:
Oh, just one hint you might not know and that I just learnt it a few days ago. All the Flash Messages are stored in the session, and whenever a new page is loaded, the flash message is displayed and then removed from the session.

Each flash message has an id, which is the first parameter for the `#addFlash()` method. If we always use `success`, we might face a situation where a flash message is shown in a wrong page. This happened to me a few weeks ago. 😄 

So, whenever we use `#addFlash`, let's give the message a unique id. I started using a naming pattern such as `controller#id` (e.g. `installment#paid_successfully` for `Controller\Installment`).